### PR TITLE
catalog-backend-module-puppetdb fixed to use puppetdb 7 api references

### PR DESCRIPTION
**Pull request by jagexjake**
## Hey, I just made a Pull Request!

This was identified in: https://github.com/backstage/backstage/issues/25224

Current implementation of this plugin was against puppetdb 6 which is no longer supported, and the latest_report_status parameter is part of the node api instead of the factsets api.
I have changed to do the discovery over the nodes api, and then for each node it will look up the factsets and add it to the factset type instead of othe node object.
This is to match default puppetdb setups https://www.puppet.com/docs/puppetdb/7/api/query/v4/nodes.html 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by jagexjake** [comment]
@freben are you able to review when you get a chance please?

File: plugins/catalog-backend-module-puppetdb/src/puppet/types.ts [hunk 1]
```diff
@@ -54,19 +54,11 @@ export type PuppetNode = {
   /**
    *  The status of the latest report. Possible values come from Puppet's report status failed, changed, or unchanged.
    */
-  hash: string;
+  latest_report_hash: string;
```
Comment (freben) [inline plugins/catalog-backend-module-puppetdb/src/puppet/types.ts:57]: the comments attached to the fields seem to not match the content of the fields

**Review by freben — COMMENTED** [review]
Just double checking, does this mean that we now drop puppet 6 compatibility? Then the changeset might be good to start the changeset text with `**BREAKING**: Dropping PuppetDB 6 compatibility` or so.

File: .changeset/cool-beans-itch.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-backend-module-puppetdb': minor
+---
+
+Change puppetdb catalog plugin to use the nodes api for catalog discovery
```
[the remark this answers is not in the document]
Comment (jagexjake) [reply .changeset/cool-beans-itch.md:5]: Puppet 6 was EOL in February 2023 https://www.puppet.com/docs/puppet/7/platform_lifecycle.html
Good idea though, will modify later on

**Pull request closed without merging**